src/librustc_driver/args/mod.rs

```diff
@@ -0,0 +1,105 @@
+use std::env;
+use std::error;
+use std::fmt;
+use std::fs;
+use std::io::{self, BufRead};
+use std::str;
+use std::sync::atomic::{AtomicBool, Ordering};
+
+#[cfg(test)]
+mod tests;
+
+static USED_ARGSFILE_FEATURE: AtomicBool = AtomicBool::new(false);
+
+pub fn used_unstable_argsfile() -> bool {
+    USED_ARGSFILE_FEATURE.load(Ordering::Relaxed)
+}
+
+struct FileArgs {
+    path: String,
+    input: Vec<u8>,
+}
+
+impl FileArgs {
+    fn new(path: String, input: Vec<u8>) -> Self {
+        FileArgs { path, input }
+    }
+
+    fn lines(self) -> impl Iterator<Item = Result<String, Error>> {
+        let Self { input, path } = self;
+        io::Cursor::new(input).lines().map(move |res| {
+            let path = path.clone();
+            res.map_err(move |err| match err.kind() {
+                io::ErrorKind::InvalidData => Error::Utf8Error(Some(path)),
```

Comment:
This is sort of why I wasn't a huge fan of using `BufRead`, the error here is impossible? This seems like some odd error handling code for a case that doesn't happen

Reply:
I don't think its impossible - it will happen if the file is non Utf8. This is just making the error clearer rather than remaining wrapped in an io::Error.